apps/src/javalab/JavalabEditor.jsx

```diff
@@ -313,6 +314,13 @@ class JavalabEditor extends React.Component {
 
     if (!filename) {
       errorMessage = javalabMsg.missingFilenameError();
+    } else if (
+      filename === '.java' ||
+      (filename.toLowerCase().endsWith('.java') && !filename.endsWith('.java'))
+    ) {
+      // if filename is either only '.java' or ends with a non-lowercase casing of '.java',
+      // give an error with an example Java filename.
+      errorMessage = javalabMsg.invalidJavaFilenameFormat();
     } else if (filename.endsWith('.java') && /\s/g.test(filename)) {
```

Comment:
Could we potentially combine all these checks (lowercase .java, no spaces, can't be just .java) into a single regex statement (e.g. something like `/^([a-zA-Z]+\.java)$/g.test(filename)`)? Might make the code here cleaner, but I also am okay with this since it allows us to provide different error messages per error type.

Reply:
I think I want to stick to the different error messages since we already had the existing one for no spaces